ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Node.hs

```diff
@@ -293,7 +294,8 @@ protocolInfoCardano genesisByron mSigThresh pVer sVer mbCredsByron
                   WrapChainDepState $
                     headerStateChainDep initHeaderStateByron
           }
-      , pInfoBlockForging = blockForging
+      , pInfoBlockForging =
+          zipWithThese mkBlockForging credssByron credssShelley
```

Comment:
What when the next era comes?

New approach: instead of `OptNP 'False` use `NS`. So when we have a block forging for Byron and Shelley, we just spawn two threads.